clients/clearly-defined/src/main/kotlin/Extensions.kt

```diff
@@ -27,3 +29,9 @@ import okio.Buffer
  * Return the request body as a string, see https://github.com/square/okhttp/issues/1891.
  */
 fun RequestBody.string() = Buffer().also { writeTo(it) }.readUtf8()
+
+/**
+ * Return this string lower-cased except for the first character which is upper-cased.
+ */
+fun String.titlecase() =
+    lowercase().replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }
```

Comment:
I was simply following the advice from the deprecation message, see

https://github.com/JetBrains/kotlin/blob/e927764aafc4571373bd2053497f8179e3cf5911/libraries/stdlib/jvm/src/kotlin/text/StringsJVM.kt#L721

I'm not sure whether using `Char.titlecaseChar()` is the right approach here, as it's documented as "This function performs one-to-one character mapping. To support one-to-many character mapping use the titlecase function", where as the `String.lowercase()` function is documented as "This function supports one-to-many and many-to-one character mapping, thus the length of the returned string can be different from the length of the original string".

So with your proposal one implementation would support many-to-one mappings while the other would not. However, I wonder whether instead of `Locale.getDefault()` rather `Locale.ROOT` should be used, as in the implementation of `String.lowercase()`. Any opinion on that?

Reply:
This Unicode stuff is indeed confusing. I would then indeed prefer `Locale.ROOT` here; with the default locale, the results of these functions can vary depending on the local setup.